app/models/collection_search_builder.rb

```diff
@@ -64,7 +64,7 @@ def apply_gated_discovery(solr_parameters, permission_types = discovery_permissi
 
     # Copied from SearchBuilder
     def only_published_items(solr_parameters, ability = current_ability)
-      if ability.cannot? :create, MediaObject
```

Comment:
`can? :create, MediaObject` is equivalent to `is_member_of_any_collection?`
See https://github.com/avalonmediasystem/avalon/blob/1ab561e38327afa172fcb0f62fe8d6a2f33cf283/app/models/ability.rb#L71-L73